cql3/statements/select_statement.cc

```diff
@@ -439,9 +439,25 @@ generate_base_key_from_index_pk(const partition_key& index_pk, const std::option
 
     for (const column_definition& base_col : base_columns) {
         const column_definition* view_col = view_schema.view_info()->view_column(base_col);
+        if (!view_col) {
+            throw std::runtime_error(format("Base key column not found in the view: {}", base_col.name_as_text()));
+        }
+        if (base_col.type != view_col->type) {
+            throw std::runtime_error(format("Mismatched types for base and view columns {}: {} and {}",
+                    base_col.name_as_text(), base_col.type->cql3_type_name(), view_col->type->cql3_type_name()));
+        }
         if (view_col->is_partition_key()) {
             exploded_base_key.push_back(index_pk.get_component(view_schema, view_col->id));
-        } else if (index_ck) {
+        } else {
+            if (!view_col->is_clustering_key()) {
+                throw std::runtime_error(
+                        format("Base primary key column is not a primary key column in the materialized view: {} (kind: {})",
```

Comment:
s/primary/clustering/?

Reply:
Both are correct, will be unique in the logs, and are only expected to be triggered by a bug in Scylla - so I think the current message is also fine. If you prefer to switch to `clustering`, let me know and I'll update the patch.

Reply:
The message is correct because the first if checked is_partition_key() and the second is_clustering_key(). If we get to print this message, it means it is neither.

Reply:
hmm, it seems confusing to me because you check for `!view_col->is_clustering_key()`

Reply:
But it follows the other if(view_col->is_partition_key())), if that other if() succeeded we wouldn't even get here...

Reply:
@bhalevy only because it's the `else` branch of `view_col->is_partition_key()`, so it's already implied that it wasn't found in the partition key

Reply:
thanks